sensor/common/sensor/central_sender_impl.go

```diff
@@ -58,7 +59,10 @@ func (s *centralSenderImpl) send(stream central.SensorService_CommunicateClient,
 		s.finished.Done()
 	}()
 
-	wrappedStream := metrics.NewSizingEventStream(stream)
+	bufferedC := make(chan *central.MsgFromSensor, env.ResponsesChannelBufferSize.IntegerSetting())
+	defer close(bufferedC)
+	wrappedStream, streamErrC := NewBufferedStream(stream, bufferedC, s.stopper.LowLevel().GetStopRequestSignal())
+	wrappedStream = metrics.NewSizingEventStream(wrappedStream)
```

Comment:
Why don't we create `bufferedC` inside `NewBufferedStream`? 
I guess, we do not use it here in it's pure form, so it would be fine hiding it.
That would prevent our future us of trying to write something in that channel later outside of the `NewBufferedStream`. I think, it would be safer this way.

Reply:
The idea is to defer the closing to the channel to the writer. I know that technically the `buffered_stream` writes to the channel but it's just through the `Send` function that ultimately is called from the `centralSenderImpl` and should only be called from there.

If we hide the channel, we need to make sure we add a `Stop` function to the stream to close the channel from the `centralSenderImpl`. I'm ok with both implementations.

Reply:
oh, that makes sense. I thought that maybe closing the `wrappedStream` could internally close the `bufferedC` - I should have checked it before commenting.